pca.go

```diff
@@ -0,0 +1,79 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// PrincipalComponents returns the principal component direction vectors of
```

Comment:
I agree you should note the behaviour about centring and scaling to variance 1.

Also @btracey I think I can help clarify the singular value thing.
The v -> f*v^2 results from the fact we are calculating the singular values of 
the data matrix a. This is sort of briefly covered on the wiki page:
https://en.wikipedia.org/wiki/Principal_component_analysis#Singular_value_decomposition
The standard formulation of PCA instead involves the 
eigendecomposition of the covariance matrix, (1/(n-1)) * a^T * a (assuming the 
columns of a are centred). The eigenvalues of this covariance matrix will be 
directly related to the singular values of a by the above transformation. This can 
be seen reasonably easily, if the singular value decomposition of a is written:
a = U * S * V^T
then the covariance matrix becomes
(1/(n-1)) * a^T * a  = (1/(n-1)) * (U * S * V^T)^T * (U * S * V^T) =  V * ((1/(n-1)) * S^2) * V^T
I cant find a handy link to this, but it is explained in several multivariate 
stats textbooks, including this one:
http://www.cambridge.org/au/academic/subjects/statistics-probability/statistical-theory-and-methods/analysis-multivariate-and-high-dimensional-data?format=HB
Hopefully that helps?